core/app/models/spree/line_item.rb

```diff
@@ -149,6 +149,11 @@ def tax_category_id
       super || variant_tax_category_id
     end
 
+    def recalculate_price
```

Comment:
@JustShah did you try if we really need this? change of user should automatically trigger recalculations.